BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Authentication
+{
+    using System;
+    using System.Reflection;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <remarks>
+    /// Our SDK currently targets net452, net46, and netstandard2.0.
+    /// Azure.Core.TokenCredential is only available for netstandard2.0.
+    /// I'm introducing this class as a wrapper so we can receive an instance of this class and pass it around within our SDK.
+    /// </remarks>
+    internal class ReflectionCredentialEnvelope : ICredentialEnvelope
+    {
+        private readonly object tokenRequestContext;
+        private readonly MethodInfo getTokenAsyncMethod;
+        private readonly MethodInfo getTokenMethod;
+        //private readonly Type accessTokenType;
```

Comment:
What was this `accessTokenType` intended for?

Reply:
I want to use reflection to call `public async ValueTask<AccessToken> GetTokenAsync`.
In this case, `TokenCredential`, `AccessToken`, and `ValueTask<>` are all unavailable to me in older frameworks.
The example code is also commented out below at Line 68.